public/app/features/dashboard/state/runRequest.ts

```diff
@@ -169,9 +169,9 @@ export function getProcessedDataFrames(results?: DataQueryResponseData[]): DataF
   for (const result of results) {
     const dataFrame = guessFieldTypes(toDataFrame(result));
 
-    // clear out any cached calcs
+    // clear out the cached info
     for (const field of dataFrame.fields) {
-      field.calcs = null;
+      field.state = undefined;
```

Comment:
nit: using `undefined` signals that this prop doesn't exists imo, `null` just states this prop is not initialized.
```suggestion
      field.state = null;
```

Reply:
So should the Field type then be:

state?: FieldState | null; 

then we have another case to think about, hate undefined! 

Another alternative is:
delete field.state;

Reply:
made it nullable, and set to null when cleared, think it makes more sense!